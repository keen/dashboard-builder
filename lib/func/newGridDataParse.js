const newGridDataParse = (dashboardInfo, chartId) => {
  const { items } = dashboardInfo.data;
  const newGridItems = items.map(chart => {
    return {
      ...chart,
      i: chartId(),
      w: Math.floor(chart.width / 100),
      h: Math.floor(chart.height / 100) * 3,
      x: Math.floor(chart.left / 100),
      y: Math.floor(chart.top / 100)
    };
  });
  const newDashboardInfo = {
    ...dashboardInfo,
    settings: {
      ...dashboardInfo.settings,
      layout: [...newGridItems],
      items: [...newGridItems]
    }
  };
  return newDashboardInfo;
};

export default newGridDataParse;
