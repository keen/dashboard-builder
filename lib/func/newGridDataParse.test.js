import newGridDataParse from './newGridDataParse';

describe('newGridDataParse()', () => {
  it('should parse data correctly', () => {
    const dashboardInfo = {
      data: {
        items: [
          {
            height: 40,
            left: 20,
            text: '<p>Component</p>',
            top: 20,
            type: 'paragraph',
            width: 1160
          },
          {
            height: 2800,
            left: 620,
            chartTitle: 'Title',
            top: 80,
            savedQuery: [
              {
                label: 'Saved query',
                value: 'saved_query'
              }
            ],
            type: 'bar',
            width: 560
          }
        ]
      }
    };
    const chartUniqueId = jest
      .fn()
      .mockImplementation(() => '1433be06-9dfa-4caf-8558-f5959e0535eb')
      .mockImplementation(() => '76d83720-c703-4127-ad92-7cd37d1ea893')
      .mockImplementation(() => '7a99a8d0-bf4f-429b-8bb7-520bf069ac59');
    const parsedData = newGridDataParse(dashboardInfo, chartUniqueId);
    expect(parsedData).toMatchSnapshot();
  });
});
