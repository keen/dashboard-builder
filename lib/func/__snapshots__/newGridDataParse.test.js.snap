// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newGridDataParse() should parse data correctly 1`] = `
Object {
  "data": Object {
    "items": Array [
      Object {
        "height": 40,
        "left": 20,
        "text": "<p>Component</p>",
        "top": 20,
        "type": "paragraph",
        "width": 1160,
      },
      Object {
        "chartTitle": "Title",
        "height": 2800,
        "left": 620,
        "savedQuery": Array [
          Object {
            "label": "Saved query",
            "value": "saved_query",
          },
        ],
        "top": 80,
        "type": "bar",
        "width": 560,
      },
    ],
  },
  "settings": Object {
    "items": Array [
      Object {
        "h": 0,
        "height": 40,
        "i": "7a99a8d0-bf4f-429b-8bb7-520bf069ac59",
        "left": 20,
        "text": "<p>Component</p>",
        "top": 20,
        "type": "paragraph",
        "w": 11,
        "width": 1160,
        "x": 0,
        "y": 0,
      },
      Object {
        "chartTitle": "Title",
        "h": 84,
        "height": 2800,
        "i": "7a99a8d0-bf4f-429b-8bb7-520bf069ac59",
        "left": 620,
        "savedQuery": Array [
          Object {
            "label": "Saved query",
            "value": "saved_query",
          },
        ],
        "top": 80,
        "type": "bar",
        "w": 5,
        "width": 560,
        "x": 6,
        "y": 0,
      },
    ],
    "layout": Array [
      Object {
        "h": 0,
        "height": 40,
        "i": "7a99a8d0-bf4f-429b-8bb7-520bf069ac59",
        "left": 20,
        "text": "<p>Component</p>",
        "top": 20,
        "type": "paragraph",
        "w": 11,
        "width": 1160,
        "x": 0,
        "y": 0,
      },
      Object {
        "chartTitle": "Title",
        "h": 84,
        "height": 2800,
        "i": "7a99a8d0-bf4f-429b-8bb7-520bf069ac59",
        "left": 620,
        "savedQuery": Array [
          Object {
            "label": "Saved query",
            "value": "saved_query",
          },
        ],
        "top": 80,
        "type": "bar",
        "w": 5,
        "width": 560,
        "x": 6,
        "y": 0,
      },
    ],
  },
}
`;
